pkg/core/resource_graph.go

```diff
@@ -15,7 +15,9 @@ type (
 
 func NewResourceGraph() *ResourceGraph {
 	return &ResourceGraph{
-		underlying: graph.NewDirected[Resource](),
+		underlying: graph.NewDirected(func(r Resource) string {
+			return r.Id().String()
+		}),
```

Comment:
im a little bit confused on this change

Reply:
Because resources no longer conform to the `Identifiable` interface and the `Id()` signature no longer matches constructs, we need a way to specify the hashing function is a more flexible way (which is also why it's used this way in the underlying graph).